### ISSUE TITLE
Add citations (dictionary and directive)

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/curriculum#2282]
Rachel is going to create a file called **citations.json** (see https://github.com/bootstrapworld/curriculum/issues/2283):
```
"politzer2021": {
        "in-text": "Politz, Denny, and Tabak, 2021",
        "apa": "J. Politz et. al., _We're off to see the wizard - a treatise on Dorai Sitaram_"
        "public-url": "https://......."
        "private-url": "https://drive.google.com...."
}
```

Then she will modify other dictionaries (like `misconceptions.json`), to cite sources using the keys in the `citations.json` file. For example:
```
"ms-bp-1": {
    "misconception" : "Students often think that the median line actually represents the mean, causing misinterpretation of skewed datasets",
    "addressed" : false,
    "proposed-solution" : [
      ""
    ],
    "lessons" : [
      "box-plots"
    ],
    "standards": [
      "",  
      ""
    ],
    "sources": [
      "politzer2021",
      "schanzer2015"
    ]
  },
```

In the lesson plan, authors can write:
`This is a common misconception @citation{politz2021}`, which will insert the `in-text` citation field into the lesson plan. Hovering over that citation should bring up the `apa` citation using the `title=` attribute:

`<span class="citation" title="J. Politz et. al., We're off to see the wizard - a treatise on Dorai Sitaram">Politz, Denny, and Tabak, 2021</span>`

Note that the in-text string includes asciidoc formatting!

**Comment by retabak** [comment]
One more question: Is there any way for the title of the journal to be italicized in the text that appears when you hover? If this is very difficult or impossible, I don't think it's a big deal. (Proper APA citations italicize the journal title.)

**Comment by retabak** [comment]
I've got three requests:
- I sort of envisioned that the citations would be hyperlinked, so people would think to hover over them. I realize now that if we aren't linking to the articles, that will be weird. Is there any other way to make them pop a little more? Or, could we hyperlink to a BS "Works Cited" page?
- I tried putting this in the asciidoc - `@citation{cooper-shore-2008, "Cooper and Shore (2008)"}` ; when I built, I got this warning: `WARNING: Lesson plan distribution/en-us/lessons/measures-of-center/index.adoc: Undefined @citation cooper-shore-2008, "Cooper and Shore (2008)"`. I would like to be able to cite papers without necessarily using the same format of in-text citation every time. Can we build in this capability?
- Another use case that I'm hoping we can account for is when we want to cite multiple sources. Here's an example: **"Very commonly, students develop a computational understanding of mean, but not a conceptual one (Bakker, Biehler, & Konold, 2004; Pollatsek, Lima, Well, 1981)."** The request here, I guess, is to be able to use `@citation{source-1; source-2}.`
